app/src/main/java/pl/pwr/zpi/bcycle/mobile/TripCreationMapActivity.kt

```diff
@@ -0,0 +1,326 @@
+                map?.animateCamera(
+                    CameraUpdateFactory.newLatLngZoom(
+                        markerStartPoint?.position,
+                        12f
+                    )
+                )
+            }
+        }
+
+        bt_show_end.setOnClickListener {
+            if (markerFinishPoint != null) {
+                markerFinishPoint?.showInfoWindow()
+                map?.animateCamera(
+                    CameraUpdateFactory.newLatLngZoom(
+                        markerFinishPoint?.position,
+                        12f
+                    )
+                )
+            }
+        }
+    }
+
+    private fun isRequiredDataGiven(): Boolean {
+        return markerStartPoint!=null && markerFinishPoint!=null
+    }
+
+    private fun createMarkersList(): MutableList<GroupTripPoint> {
+        val list = mutableListOf<GroupTripPoint>()
+        list.add(
+            GroupTripPoint(
+                markerStartPoint!!.position.latitude,
+                markerStartPoint!!.position.longitude,
+                null
+            )
+        )
+        for (marker in myMarkers) {
+            if (marker != markerStartPoint && marker != markerFinishPoint) {
+                list.add(GroupTripPoint(marker.position.latitude, marker.position.longitude, null))
+            }
+        }
+        list.add(
+            GroupTripPoint(
+                markerFinishPoint!!.position.latitude,
+                markerFinishPoint!!.position.longitude,
+                null
+            )
+        )
+        return list
+    }
+
+    override fun onMapReady(googleMap: GoogleMap?) {
+        map = googleMap
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this)
+        map?.setOnMarkerClickListener(this)
+
+        handleMapGestures()
+        getCurrentLocation()
+
+        map?.setOnMapLongClickListener {
+            val markerOpt =
+                MarkerOptions().position(it).icon(BitmapDescriptorFactory.defaultMarker())
+            val mark = map?.addMarker(markerOpt)
+            mark?.isDraggable = true
+
+            // myMarkers.add(GroupTripPoint(markerOpt.position.latitude, markerOpt.position.longitude, null))
+            myMarkers.add(mark!!)
+        }
+    }
+
+    private fun handleMapGestures() {
+        val mapSettings = map?.uiSettings
+        mapSettings?.isZoomGesturesEnabled = true
+        mapSettings?.isScrollGesturesEnabled = true
+    }
+
+    // region markerClickListener
+    override fun onMarkerClick(marker: Marker?): Boolean {
+        val array = arrayListOf<String>(
+            getString(R.string.set_as_start_point), getString(R.string.set_as_end_point), getString(
+                R.string.remove_it
+            )
+        )
+        LovelyChoiceDialog(this)
+            .setTopColorRes(R.color.violet)
+            .setTitle(resources.getString(R.string.prompt_marker_what_to_do))
+            .setIcon(R.drawable.bike_icon)
+            .setItemsMultiChoice(
```

Comment:
A multi-choice dialog is unintuitive and lets users pick nonsense combinations (start point + remove) Use `.setItems` instead, and pass a list with five options (start, end, start & end, remove, cancel editing).